### ISSUE TITLE
Can not visit subdomain.localhost  

### ISSUE THREAD

**Issue by beocaca** [cypress-io/cypress#18881]
### Current behavior

iam trying to visit subdomain in local host but it does not work, by any site else have working ex: 'hello.example.com'
anyboy have solution please!
![image](https://user-images.githubusercontent.com/51392704/141275970-068fa538-42e9-48a9-a73d-00783ed6f08d.png)

### Desired behavior

![image](https://user-images.githubusercontent.com/51392704/141276496-413b8c4c-0f0a-4926-96a8-88f4b46a1553.png)

### Test code to reproduce

![image](https://user-images.githubusercontent.com/51392704/141276531-cb78de57-338f-4e30-88bc-4b951fcf39ae.png)

### Cypress Version

9.0

**Comment by jawira** [comment]
I just commented about this in another issue :(  https://github.com/cypress-io/cypress/issues/451#issuecomment-975569197

**tl;dr** To fix this you have to add manually `127.0.0.1 subdomain.localhost` to your `/etc/hosts` file.

related #3122 

EDIT: I think the solution proposed by @puranjayjain is better!

**Comment by puranjayjain** [comment]
I fixed it by adding to cypress config 

`config.hosts = { '*.localhost': '127.0.0.1' };`

**Comment by AtofStryker** [comment]
Hi @beocaca. Have you tried this with different hosts in `etc/hosts` for instance? `localhost` is a bit different and might not resolve subdomains correctly, but you should be able to accomplish a subdomain navigation in cypress 10+ as well as leverage the `experimentalSessionAndOrigin` flag in Cypress 10+ that allows for cross origin navigations (which likely isnt the case here). I am going to close this issue, but if you are still having problems with this please let us know!

**Comment by joaottc** [comment]
> I fixed it by adding to cypress config
> 
> `config.hosts = { '*.localhost': '127.0.0.1' };`

Can anyone confirm this still works in version 12.x? I've tried it and it doesn't seem to work anymore

**Comment by alexwhin** [comment]
> { '*.localhost': '127.0.0.1' }

I can confirm this is working for me in `^13.6.2`

```ts
import { defineConfig } from "cypress";

export default defineConfig({
  e2e: {
    ...
  },
  hosts: { "*.localhost": "127.0.0.1" },
  ...
});

```
